src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -413,7 +412,7 @@ impl<'a, 'tcx> LinkCollector<'a, 'tcx> {
                 ResolutionFailure::NotResolved {
                     module_id,
                     partial_res: Some(Res::Primitive(prim_ty)),
-                    unresolved: item_str.into(),
+                    unresolved: item_name.to_string().into(),
```

Comment:
.as_str().into() doesn't compile, because `as_str()` returns a SymbolStr, not a &'static str. So you can't return any borrow of the string after the SymbolStr is dropped.

Reply:
But `SymbolStr` should be `Deref<str>`. Weird, but fair.

Reply:
¯\_(ツ)_/¯ I guess that's what it is. Thanks for trying.